File: agentcc.py
#!/usr/bin/env python3
"""
agentcc - Claude Code Chat Simulator
A command-line tool to interact with Claude API with context memory.
"""

import os
import sys
import json
import anthropic
from pathlib import Path

# Context file location
CONTEXT_FILE = Path.home() / ".agentcc_context.json"

def load_context():
    """Load conversation context from file."""
    if CONTEXT_FILE.exists():
        try:
            with open(CONTEXT_FILE, 'r') as f:
                return json.load(f)
        except:
            return []
    return []

def save_context(context):
    """Save conversation context to file."""
    with open(CONTEXT_FILE, 'w') as f:
        json.dump(context, f)

def clear_context():
    """Clear conversation context."""
    if CONTEXT_FILE.exists():
        CONTEXT_FILE.unlink()
    print("Context cleared.")

def send_message(prompt):
    """Send message to Claude and get response."""
    # Load conversation history
    context = load_context()
    
    # Add user message to context
    context.append({
        "role": "user",
        "content": prompt
    })
    
    # Get API key from environment
    api_key = os.environ.get("ANTHROPIC_API_KEY")
    if not api_key:
        print("Error: ANTHROPIC_API_KEY environment variable not set.")
        sys.exit(1)
    
    # Initialize Claude client
    client = anthropic.Anthropic(api_key=api_key)
    
    try:
        # Send message to Claude
        response = client.messages.create(
            model="claude-3-5-sonnet-20241022",
            max_tokens=8096,
            messages=context
        )
        
        # Extract response text
        response_text = response.content[0].text
        
        # Add assistant response to context
        context.append({
            "role": "assistant",
            "content": response_text
        })
        
        # Save updated context
        save_context(context)
        
        # Print response
        print(response_text)
        
    except Exception as e:
        print(f"Error: {str(e)}")
        sys.exit(1)

def main():
    """Main entry point."""
    if len(sys.argv) < 2:
        print("Usage: agentcc \"PROMPT_MSG\"")
        print("       agentcc \"/clear\"")
        sys.exit(1)
    
    prompt = sys.argv[1]
    
    # Handle special commands
    if prompt == "/clear":
        clear_context()
        return
    
    # Send normal message
    send_message(prompt)

if __name__ == "__main__":
    main()

File: agentcc.js
#!/usr/bin/env node
/**
 * agentcc - Claude Code Chat Simulator
 * A command-line tool to interact with Claude API with context memory.
 */

const Anthropic = require('@anthropic-ai/sdk');
const fs = require('fs');
const path = require('path');
const os = require('os');

// Context file location
const CONTEXT_FILE = path.join(os.homedir(), '.agentcc_context.json');

/**
 * Load conversation context from file.
 */
function loadContext() {
    if (fs.existsSync(CONTEXT_FILE)) {
        try {
            const data = fs.readFileSync(CONTEXT_FILE, 'utf8');
            return JSON.parse(data);
        } catch (err) {
            console.error(`Warning: Could not load context file: ${err.message}`);
            return [];
        }
    }
    return [];
}

/**
 * Save conversation context to file.
 */
function saveContext(context) {
    try {
        fs.writeFileSync(CONTEXT_FILE, JSON.stringify(context, null, 2));
    } catch (err) {
        console.error(`Error: Could not save context file: ${err.message}`);
        process.exit(1);
    }
}

/**
 * Clear conversation context.
 */
function clearContext() {
    if (fs.existsSync(CONTEXT_FILE)) {
        fs.unlinkSync(CONTEXT_FILE);
    }
    console.log('Context cleared.');
}

/**
 * Send message to Claude and get response.
 */
async function sendMessage(prompt) {
    // Load conversation history
    const context = loadContext();
    
    // Add user message to context
    context.push({
        role: 'user',
        content: prompt
    });
    
    // Get API key from environment
    const apiKey = process.env.ANTHROPIC_API_KEY;
    if (!apiKey) {
        console.error('Error: ANTHROPIC_API_KEY environment variable not set.');
        process.exit(1);
    }
    
    // Initialize Claude client
    const client = new Anthropic({ apiKey });
    
    try {
        // Send message to Claude
        const response = await client.messages.create({
            model: 'claude-3-5-sonnet-20241022',
            max_tokens: 8096,
            messages: context
        });
        
        // Extract response text
        const responseText = response.content[0].text;
        
        // Add assistant response to context
        context.push({
            role: 'assistant',
            content: responseText
        });
        
        // Save updated context
        saveContext(context);
        
        // Print response
        console.log(responseText);
        
    } catch (err) {
        console.error(`Error: ${err.message}`);
        process.exit(1);
    }
}

/**
 * Main entry point.
 */
async function main() {
    if (process.argv.length < 3) {
        console.log('Usage: agentcc "PROMPT_MSG"');
        console.log('       agentcc "/clear"');
        process.exit(1);
    }
    
    const prompt = process.argv[2];
    
    // Handle special commands
    if (prompt === '/clear') {
        clearContext();
        return;
    }
    
    // Send normal message
    await sendMessage(prompt);
}

// Run main function
main().catch(err => {
    console.error(`Error: ${err.message}`);
    process.exit(1);
});
